### PR TITLE
add task solution

### PR DESCRIPTION
**Pull request by maksym-nemera**
[DEMO LINK](https://maksym-nemera.github.io/react_movies-list-add-form/)

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -1,16 +1,24 @@
...
+import { useState } from 'react';
 import { MoviesList } from './components/MoviesList';
 import { NewMovie } from './components/NewMovie';
 import moviesFromServer from './api/movies.json';
+import { Movie } from './types/Movie';
 
 export const App = () => {
+  const [movies, setMovies] = useState(moviesFromServer);
+
+  const handleAddMovie = (movie: Movie) => {
```
Comment (ruslanpashkov) [inline src/App.tsx:11]: it's not a handler, so you don't need `handle` keyword here
```suggestion
  const addMovie = (movie: Movie) => {
```

File: src/components/NewMovie/NewMovie.tsx [hunk 2]
```diff
@@ -1,45 +1,118 @@
-import { useState } from 'react';
+import { FC, useEffect, useState } from 'react';
 import { TextField } from '../TextField';
+import { Movie } from '../../types/Movie';
 
-export const NewMovie = () => {
+type Props = {
+  onAdd: (movie: Movie) => void,
```
Comment (ruslanpashkov) [inline src/components/NewMovie/NewMovie.tsx:6]:
```suggestion
  onAdd: (movie: Movie) => void;
```

File: src/components/NewMovie/NewMovie.tsx [hunk 3]
```diff
@@ -1,45 +1,118 @@
...
+      setDisabledButton(false);
+    }
+  }, [title, imgUrl, imdbUrl, imdbId]);
+
+  const resetForm = () => {
+    setTitle('');
+    setDescription('');
+    setImgUrl('');
+    setImdbUrl('');
+    setImdbId('');
```
Comment (ruslanpashkov) [inline src/components/NewMovie/NewMovie.tsx:40]:
```suggestion
    setFormData(initialFormData);
```

File: src/components/NewMovie/NewMovie.tsx [hunk 4]
```diff
@@ -1,45 +1,118 @@
...
+  };
+
+  const handleSubmit = (event: React.FormEvent) => {
+    event.preventDefault();
+
+    setCount(count + 1);
```
Comment (ruslanpashkov) [inline src/components/NewMovie/NewMovie.tsx:46]: we change our current state based on previous state, let's use functional `setCount` here

File: src/components/NewMovie/NewMovie.tsx [hunk 5]
```diff
@@ -1,45 +1,118 @@
...
+    if (count > 0) {
+      onAdd(formData as Movie);
+    }
+  }, [count]);
+
+  useEffect(() => {
+    if (title && imgUrl && imdbUrl && imdbId) {
+      setDisabledButton(false);
+    }
+  }, [title, imgUrl, imdbUrl, imdbId]);
```
Comment (ruslanpashkov) [inline src/components/NewMovie/NewMovie.tsx:33]: what if the fields are only filled with spaces?

also we can create separate variable for it and pass negation of this value to `disabled` attribute

**Review by ruslanpashkov — CHANGES_REQUESTED** [review]

**Review by silchyn — CHANGES_REQUESTED** [review]
Form can be submitted several times in a row with no values entered

**Review by silchyn — CHANGES_REQUESTED** [review]
When all values are spaces, submit button is disabled, but no validation message is shown
image

**Review by And678 — APPROVED** [review]
Good work, approve :heavy_check_mark:

**Pull request closed without merging**